tests/kernel/fpu_sharing/generic/src/pi.c

```diff
@@ -176,6 +192,24 @@ void test_pi(void)
 	k_thread_start(pi_low);
 	k_thread_start(pi_high);
 
+#if !defined(DISABLE_INT_TEST)
+	k_timer_start(&pi_timer, K_MSEC(200), K_MSEC(1));
+#endif
```

Comment:
I guess the point you want to test is "can the system take an interrupt while calculating pi and not lose state or otherwise corrupt the FPU state of the interrupted thread?".

If that's the case, then maybe you don't want to be doing pi in the interrupt and instead do something complicated and synchronous with lots of FPU registers?  The code as it is is only doing four operations on three static values.